### PR TITLE
feat(ui): checkbox component

### PR DESCRIPTION
**Pull request by artyorsh**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:

implements `CheckBox` component

### CODE REVIEW

File: src/framework/ui/checkbox/checkbox.component.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  Interaction,
+} from '@kitten/theme';
+import { CheckMark } from '@kitten/ui/drawable/checkmark/checkmark.component';
+
+interface CheckBoxProps {
+  checked?: boolean;
+  onChange?: (checked: boolean) => void;
+  appearance?: string | 'default';
+  status?: string | 'error';
+  size?: string | 'big' | 'small';
```
Comment (malashkevich) [inline src/framework/ui/checkbox/checkbox.component.tsx]: Please remove exact values

File: src/framework/ui/checkbox/checkbox.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  it('* emits onChange with correct args', () => {
+    const onChange = jest.fn();
+    const component = renderComponent({
+      checked: true,
+      onChange: onChange,
+    });
+    fireEvent.press(component.getByType(TouchableOpacity));
+
+    expect(onChange).toBeCalledWith(true);
```
Comment (malashkevich) [inline src/framework/ui/checkbox/checkbox.spec.tsx]: here should be `false`

File: src/framework/ui/checkbox/checkbox.component.tsx (hunk 1 above)
Comment (artyorsh) [reply src/framework/ui/checkbox/checkbox.component.tsx]: currently the same syntax is used in [radio component](https://github.com/akveo/react-native-ui-kitten/blob/next/src/framework/ui/radio/radio.component.tsx#L14)
we can refactor this in each implemented component in future

File: src/framework/ui/drawable/checkmark/checkmark.component.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import {
+  View,
+  ViewProps,
+} from 'react-native';
+
+type Props = ViewProps;
+
+// TODO(ui): checkmark component
+
+export class CheckMark extends React.Component<Props> {
```
Comment (32penkin) [inline src/framework/ui/drawable/checkmark/checkmark.component.tsx]: There is such demo component in the "toggle" PR

**Pull request merged**